homeassistant/components/myio/__init__.py

```diff
@@ -0,0 +1,120 @@
+
+PLATFORMS = [SENSOR]
+
+COMMS_THREAD = CommsThread()
+
+
+async def async_setup(hass, config):
+    """Set up the myIO-server component."""
+    return True
+
+
+async def async_setup_entry(hass, config_entry):
+    """Set up config entry."""
+    _server_name = slugify(config_entry.data[CONF_NAME])
+    _LOGGER.debug(config_entry)
+
+    hass.data.setdefault(DOMAIN, {})[_server_name] = {}
+
+    hass.data[DOMAIN][_server_name]["state"] = "Offline"
+
+    try:
+        _refresh_timer = config_entry.options[CONF_REFRESH_TIME]
+    except (ValueError, Exception):  # pylint: disable=broad-except
+        _refresh_timer = 4
+
+    def server_data():
+        """Return the server data dictionary database."""
+        return hass.data[DOMAIN][_server_name]
+
+    async def setup_config_entries():
+        """Set up myIO platforms with config entry."""
+
+        for component in PLATFORMS:
+            hass.async_create_task(
+                hass.config_entries.async_forward_entry_setup(config_entry, component)
+            )
+
+    async def async_update_data():
+        """Fetch data from API endpoint."""
+        was_offline = False
+        _temp_coordinator = hass.data[DOMAIN][_server_name]["coordinator"]
+        _temp_server_state = hass.data[DOMAIN][_server_name]["state"]
+
+        del hass.data[DOMAIN][_server_name]["coordinator"]
+        del hass.data[DOMAIN][_server_name]["state"]
+
+        if _temp_server_state == "Offline":
+            hass.states.async_set(f"{_server_name}.available", False)
+            was_offline = True
+
+        # Pull fresh data from server.
+        [hass.data[DOMAIN][_server_name], _temp_server_state] = await COMMS_THREAD.send(
+            server_data=server_data(),
+            server_status=_temp_server_state,
+            config_entry=config_entry,
+            _post=None,
+        )
+
+        hass.states.async_set(f"{_server_name}.state", _temp_server_state)
+        hass.data[DOMAIN][_server_name]["state"] = _temp_server_state
+
+        if _temp_server_state.startswith("Online") and was_offline:
+            hass.states.async_set(f"{_server_name}.available", True)
+            await setup_config_entries()
+        elif not was_offline and _temp_server_state == "Offline":
+            hass.data[DOMAIN][_server_name]["coordinator"] = _temp_coordinator
+            raise PlatformNotReady
+
+        hass.data[DOMAIN][_server_name]["coordinator"] = _temp_coordinator
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        # Name of the data. For logging purposes.pip
+        name=f"{_server_name} status",
+        update_method=async_update_data,
+        # Polling interval.
+        update_interval=timedelta(seconds=_refresh_timer),
+    )
+
+    def listener():
+        """Listen to coordinator."""
+
+    hass.data[DOMAIN][_server_name]["coordinator"] = coordinator.data
+
+    await coordinator.async_refresh()
+
+    coordinator.async_add_listener(listener)
```

Comment:
`listener()` does nothing?

Reply:
yes... but it needed, to run the coordinator... If there is no listener coordinator not refreshing the database.